src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -1215,6 +1272,81 @@ public function delete_temp_backup() {
 
 		return $errors->has_errors() ? $errors : true;
 	}
+
+	/**
+	 * Performs two loopback requests, one to the dashboard and one to the homepage,
+	 * to check for potential fatal errors.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return bool Whether a fatal error was detected.
+	 */
+	protected function fatal_error_detected() {
+		$scrape_key   = md5( rand() );
+		$transient    = 'scrape_key_' . $scrape_key;
+		$scrape_nonce = (string) rand();
```

Comment:
@costdev could we use `wp_rand` here?

Reply:
We could, though this uses [the same approach](https://github.com/wordpress/wordpress-develop/blob/6a619851c27b4103d7abba5c0a047e3f68c793a6/src/wp-admin/includes/file.php#L529) that the Plugin File Editor and Theme File Editor use to generate the transient and nonce. I'm not sure that using `wp_rand()` in this context will provide any benefits.